Toplean_Alexandru/week-04/Domain/Domain/CreateClientOp/CreateClientOp.cs

```diff
@@ -8,21 +8,37 @@
 using static Domain.Domain.CreateClientOp.CreateClientResult;
 using Domain.Models;
 using Persistence.EfCore;
+using Domain.Queries;
+using Persistence;
 
 namespace Domain.Domain.CreateClientOp
 {
     public class CreateClientOp : OpInterpreter<CreateClientCmd, ICreateClientResult, Unit>
     {
-        //To Be Implemented
-        public override Task<ICreateClientResult> Work(CreateClientCmd Op, Unit state)
+        private readonly LiveInterpreterAsync interpreter;
+
+        public CreateClientOp(LiveInterpreterAsync interpret)
+        {
+            interpreter = interpret;
+        }
+
+        public async override Task<ICreateClientResult> Work(CreateClientCmd Op, Unit state)
         {
             try
             {
-                return Task.FromResult<ICreateClientResult>(new ClientCreated(new ClientAgg(Op.Client)));
+                var query = await interpreter.Interpret(Database.Query<GetClientQuery, Client>(new GetClientQuery(Op.Client.Username)), Unit.Default);
```

Comment:
If you actually need that client here, that should come as an input in your command. I'd expect your expression to be composed of the following steps:

1. query for existing client
2. create the business related entities out of that (the outcome from step 1 should be either: a) you got an existing client or b) that client doesn't exist, therefore your business entities should reflect that (the fact that's a new clinet)
3. AddOrUpdate.